service/storage_service.cc

```diff
@@ -3118,7 +3120,11 @@ future<> storage_service::replicate_to_all_cores(mutable_token_metadata_ptr tmpt
                 } else {
                     erm = pending_effective_replication_maps[this_shard_id()][table->schema()->ks_name()];
                 }
-                pending_table_erms[this_shard_id()].emplace(id, std::move(erm));
+                if (table->schema()->is_view()) {
+                    pending_view_erms[this_shard_id()].emplace(id, std::move(erm));
+                } else {
+                    pending_table_erms[this_shard_id()].emplace(id, std::move(erm));
+                }
```

Comment:
Is this loop not a huge stall with large table counts (independently of the patch)? Or with a single table with large numbers of tablets?

Reply:
Good question, we haven't see an stalls in our tests, but maybe we didn't test with a large number of tablet-based tables.
`make_replication_map` can take significant amount of cpu (like `create_effective_replication_map`), and we should allow yielding in the loop as well.

Reply:
Sorry, after looking deeper into the code, I realized that `make_replication_map` just allocates and constructs a `tablet_effective_replication_map` using the shared replication_strategy_ptr and token_metadata_ptr so it is not cpu intensive, though allocating thousands of objects via `tablet_aware_replication_strategy::do_make_replication_map` might stall as well.

Reply:
https://github.com/scylladb/scylladb/pull/21924